# CS 4662: ADVANCED MACHINE LEARNING
## Jay Tandel

### Problem1: Diagnosis of Fetal Cardio-Vascular Disease from Cardiotocography Using ANN in SciKitLearn:

### a)

In this problem, I selected a new medical dataset from UCI ML repository. The dataset includes 2126 fetal cardiotocograms (CTGs) collected in the Biomedical Engineering Institute in Porto, Portugal. The dataset includes 21 feature elements, and the output (NSP) has 3 classes: 1: Normal; 2: Suspect; 3: Pathologic.

For more info, you can visit:  https://archive.ics.uci.edu/ml/datasets/Cardiotocography

You can download the clean version of the data from github using the following command, and assign it to a Pandas DataFrame:
df  = pd.read_csv("https://github.com/mpourhoma/CS4662/raw/master/CTG_clean.csv")

The goal is to build an Artificial Neural Network that can diagnose (classify) the fetal health status. The last column (NSP) in the dataset is the label: NSP - fetal state class code (1=normal; 2=suspect; 3=pathologic), and the first 21 column are the features. To see what each feature represents, take a look at the dataset info link above.

In [1]:
from sklearn.neural_network import MLPClassifier

from sklearn import preprocessing 
from sklearn.model_selection import cross_val_score
from sklearn.metrics import accuracy_score
from sklearn.model_selection import GridSearchCV
import numpy as np
import pandas as pd

In [2]:
# Reading the csv file from web, and store it in panda DataFrame
df = pd.read_csv('https://github.com/mpourhoma/CS4662/raw/master/CTG_clean.csv')

#print the imported dataset
print(df)

       LB     AC     FM     UC     DL   DS   DP  ASTV  MSTV  ALTV  ...  Min  \
0     120  0.000  0.000  0.000  0.000  0.0  0.0    73   0.5    43  ...   62   
1     132  0.006  0.000  0.006  0.003  0.0  0.0    17   2.1     0  ...   68   
2     133  0.003  0.000  0.008  0.003  0.0  0.0    16   2.1     0  ...   68   
3     134  0.003  0.000  0.008  0.003  0.0  0.0    16   2.4     0  ...   53   
4     132  0.007  0.000  0.008  0.000  0.0  0.0    16   2.4     0  ...   53   
...   ...    ...    ...    ...    ...  ...  ...   ...   ...   ...  ...  ...   
2121  140  0.000  0.000  0.007  0.000  0.0  0.0    79   0.2    25  ...  137   
2122  140  0.001  0.000  0.007  0.000  0.0  0.0    78   0.4    22  ...  103   
2123  140  0.001  0.000  0.007  0.000  0.0  0.0    79   0.4    20  ...  103   
2124  140  0.001  0.000  0.006  0.000  0.0  0.0    78   0.4    27  ...  103   
2125  142  0.002  0.002  0.008  0.000  0.0  0.0    74   0.4    36  ...  117   

      Max  Nmax  Nzeros  Mode  Mean  Median  Varian

In [3]:
# Creating the Feature Matrix for Cancer dataset:
features = ['LB', 'AC', 'FM', 'UC', 'DL', 'DS', 'DP', 'ASTV', 'MSTV', 'ALTV', 'MLTV', 'Width', 
            'Min', 'Max', 'Nmax', 'Nzeros', 'Mode', 'Mean', 'Median', 'Variance', 'Tendency']

X = df[features]
y = df['NSP']

# print(X.head())
# print(y.head())

### b)

Use preprocessing.scale function to normalize the features.

In [4]:
# normalize the features
X_scaled = preprocessing.scale(X)
print(X_scaled)

[[-1.35222005 -0.8223883  -0.20320955 ... -1.18164215  1.87056871
   1.11298001]
 [-0.1325256   0.73013282 -0.20320955 ...  0.13203796 -0.23499819
  -0.52452553]
 [-0.03088439 -0.04612774 -0.20320955 ... -0.00624416 -0.2004807
  -0.52452553]
 ...
 [ 0.68060404 -0.56363478 -0.20320955 ...  0.96173066 -0.51113811
   1.11298001]
 [ 0.68060404 -0.56363478 -0.20320955 ...  0.8925896  -0.51113811
   1.11298001]
 [ 0.88388645 -0.30488126 -0.16034157 ...  0.47774325 -0.61469058
  -0.52452553]]


### c)

Design an ANN with one hidden layer with 30 neurons to predict the fetal’s health status. For your ANN, Use random_state=1, learning_rate_init = 0.02, solver='adam', alpha=1, verbose=True, activation='logistic'. Use 10-fold Cross-Validation to evaluate your model. Make sure to add “verbose=True” to see the training process. Then, Test your ANN on testing set, and calculate and report the accuracy.

In [5]:
# define ANN 
my_ANN = MLPClassifier(hidden_layer_sizes=(30,), activation= 'logistic', 
                       solver='adam', alpha=1, random_state=1, 
                       learning_rate_init = 0.02, verbose=True)

In [6]:
# Calculate Accuracy
accuracy_list = cross_val_score(my_ANN, X_scaled, y, cv=10, scoring='accuracy')

Iteration 1, loss = 0.73005878
Iteration 2, loss = 0.47142985
Iteration 3, loss = 0.40348082
Iteration 4, loss = 0.37549356
Iteration 5, loss = 0.35592048
Iteration 6, loss = 0.34466160
Iteration 7, loss = 0.33840068
Iteration 8, loss = 0.33539443
Iteration 9, loss = 0.33197606
Iteration 10, loss = 0.32828208
Iteration 11, loss = 0.32746580
Iteration 12, loss = 0.33105016
Iteration 13, loss = 0.32745685
Iteration 14, loss = 0.32315432
Iteration 15, loss = 0.32149350
Iteration 16, loss = 0.31980687
Iteration 17, loss = 0.32229616
Iteration 18, loss = 0.31847519
Iteration 19, loss = 0.32140209
Iteration 20, loss = 0.32571216
Iteration 21, loss = 0.31710764
Iteration 22, loss = 0.31844087
Iteration 23, loss = 0.31737602
Iteration 24, loss = 0.31580114
Iteration 25, loss = 0.31797280
Iteration 26, loss = 0.31711533
Iteration 27, loss = 0.31843518
Iteration 28, loss = 0.31552199
Iteration 29, loss = 0.31467060
Iteration 30, loss = 0.32010024
Iteration 31, loss = 0.31872513
Iteration 32, los

Iteration 43, loss = 0.33987600
Iteration 44, loss = 0.34080296
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.72549750
Iteration 2, loss = 0.48061996
Iteration 3, loss = 0.40966656
Iteration 4, loss = 0.38219590
Iteration 5, loss = 0.36690129
Iteration 6, loss = 0.35535672
Iteration 7, loss = 0.34975599
Iteration 8, loss = 0.34220654
Iteration 9, loss = 0.33872223
Iteration 10, loss = 0.33580974
Iteration 11, loss = 0.33239546
Iteration 12, loss = 0.33872517
Iteration 13, loss = 0.33028473
Iteration 14, loss = 0.33422687
Iteration 15, loss = 0.33317632
Iteration 16, loss = 0.33474331
Iteration 17, loss = 0.33118191
Iteration 18, loss = 0.33040775
Iteration 19, loss = 0.32641868
Iteration 20, loss = 0.33132205
Iteration 21, loss = 0.32603688
Iteration 22, loss = 0.32691103
Iteration 23, loss = 0.32502463
Iteration 24, loss = 0.34971755
Iteration 25, loss = 0.32997516
Iteration 26, loss = 0.33525771
Iteration 27, loss = 0.

In [7]:
# print accuracy
print(accuracy_list)

[0.84507042 0.78873239 0.88732394 0.90610329 0.87323944 0.90140845
 0.86792453 0.90566038 0.85849057 0.67924528]


### d)

 Fix the random state for reproducibility using this command:  seed = 1,  np.random.seed(seed).     

Now, use GridSearchCV to find the best number of neurons for your 1-hidden layer network. Search in the range of (5-250) with the step size of 5 for the number of neurons (i.e. 5 neurons, 10 neurons, 15 neurons, … , 245 neurons, 250 neurons). As for other parameters, use the same arguments as part (c) for your network. What is the best accuracy, and best number of neurons?

In [10]:
seed = 1
np.random.seed(seed)

# define a range for the "number of neurons" in the hidden layer for a network with 1 hidden layer:
neuron_number = [(i,) for i in range(5,251,5)]

# create a dictionary for grid parameter:
param_grid = dict(hidden_layer_sizes = neuron_number)
print(param_grid,'\n')

# creat the grid, and define the metric for evaluating the model: 
grid = GridSearchCV(my_ANN, param_grid, cv=10, scoring='accuracy', verbose=True)

# fit the grid (start the grid search):
grid.fit(X_scaled, y)

{'hidden_layer_sizes': [(5,), (10,), (15,), (20,), (25,), (30,), (35,), (40,), (45,), (50,), (55,), (60,), (65,), (70,), (75,), (80,), (85,), (90,), (95,), (100,), (105,), (110,), (115,), (120,), (125,), (130,), (135,), (140,), (145,), (150,), (155,), (160,), (165,), (170,), (175,), (180,), (185,), (190,), (195,), (200,), (205,), (210,), (215,), (220,), (225,), (230,), (235,), (240,), (245,), (250,)]} 

Fitting 10 folds for each of 50 candidates, totalling 500 fits
Iteration 1, loss = 1.18592117
Iteration 2, loss = 0.68253910
Iteration 3, loss = 0.53575745
Iteration 4, loss = 0.48467409
Iteration 5, loss = 0.45629483
Iteration 6, loss = 0.43763624
Iteration 7, loss = 0.42417208
Iteration 8, loss = 0.41228641
Iteration 9, loss = 0.40305781
Iteration 10, loss = 0.39556078
Iteration 11, loss = 0.38842024
Iteration 12, loss = 0.38271321
Iteration 13, loss = 0.37766281
Iteration 14, loss = 0.37292978
Iteration 15, loss = 0.36915124
Iteration 16, loss = 0.36621130
Iteration 17, loss = 0.3647

[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.


Iteration 48, loss = 0.33210263
Iteration 49, loss = 0.33207094
Iteration 50, loss = 0.33223243
Iteration 51, loss = 0.33135502
Iteration 52, loss = 0.33294795
Iteration 53, loss = 0.33146028
Iteration 54, loss = 0.33110900
Iteration 55, loss = 0.33223309
Iteration 56, loss = 0.33143004
Iteration 57, loss = 0.33150153
Iteration 58, loss = 0.33164772
Iteration 59, loss = 0.33037149
Iteration 60, loss = 0.33073915
Iteration 61, loss = 0.33525875
Iteration 62, loss = 0.33128957
Iteration 63, loss = 0.32914639
Iteration 64, loss = 0.33103618
Iteration 65, loss = 0.32920153
Iteration 66, loss = 0.32944103
Iteration 67, loss = 0.32968875
Iteration 68, loss = 0.33114740
Iteration 69, loss = 0.33039357
Iteration 70, loss = 0.32745477
Iteration 71, loss = 0.33067400
Iteration 72, loss = 0.32961491
Iteration 73, loss = 0.32934228
Iteration 74, loss = 0.32876822
Iteration 75, loss = 0.32801615
Iteration 76, loss = 0.32840328
Iteration 77, loss = 0.32746305
Iteration 78, loss = 0.32869655
Iteratio

Iteration 97, loss = 0.33477540
Iteration 98, loss = 0.33511410
Iteration 99, loss = 0.33549568
Iteration 100, loss = 0.33313687
Iteration 101, loss = 0.33281377
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 1.18978735
Iteration 2, loss = 0.69210145
Iteration 3, loss = 0.54576077
Iteration 4, loss = 0.49610304
Iteration 5, loss = 0.47131988
Iteration 6, loss = 0.45339267
Iteration 7, loss = 0.43861860
Iteration 8, loss = 0.42882506
Iteration 9, loss = 0.41919178
Iteration 10, loss = 0.41090811
Iteration 11, loss = 0.40378383
Iteration 12, loss = 0.39835644
Iteration 13, loss = 0.39173299
Iteration 14, loss = 0.38827907
Iteration 15, loss = 0.38217866
Iteration 16, loss = 0.37901334
Iteration 17, loss = 0.37599716
Iteration 18, loss = 0.37327361
Iteration 19, loss = 0.37255604
Iteration 20, loss = 0.36929327
Iteration 21, loss = 0.36825731
Iteration 22, loss = 0.36379426
Iteration 23, loss = 0.36214406
Iteration 24, loss = 

Iteration 6, loss = 0.45197586
Iteration 7, loss = 0.43843607
Iteration 8, loss = 0.42697444
Iteration 9, loss = 0.41762351
Iteration 10, loss = 0.41080799
Iteration 11, loss = 0.40273701
Iteration 12, loss = 0.39661784
Iteration 13, loss = 0.38999516
Iteration 14, loss = 0.38693365
Iteration 15, loss = 0.38410342
Iteration 16, loss = 0.37734679
Iteration 17, loss = 0.37475598
Iteration 18, loss = 0.37231740
Iteration 19, loss = 0.37056131
Iteration 20, loss = 0.36808148
Iteration 21, loss = 0.36479623
Iteration 22, loss = 0.36493242
Iteration 23, loss = 0.36300877
Iteration 24, loss = 0.36084819
Iteration 25, loss = 0.35872558
Iteration 26, loss = 0.35797734
Iteration 27, loss = 0.35676216
Iteration 28, loss = 0.35736534
Iteration 29, loss = 0.35611826
Iteration 30, loss = 0.35434225
Iteration 31, loss = 0.35484429
Iteration 32, loss = 0.35469748
Iteration 33, loss = 0.35235557
Iteration 34, loss = 0.35317120
Iteration 35, loss = 0.35349045
Iteration 36, loss = 0.35140535
Iteration 37

Iteration 34, loss = 0.32146607
Iteration 35, loss = 0.31839224
Iteration 36, loss = 0.31698848
Iteration 37, loss = 0.31850437
Iteration 38, loss = 0.31901690
Iteration 39, loss = 0.32053512
Iteration 40, loss = 0.31919962
Iteration 41, loss = 0.31665510
Iteration 42, loss = 0.31679366
Iteration 43, loss = 0.31636070
Iteration 44, loss = 0.31720301
Iteration 45, loss = 0.31703826
Iteration 46, loss = 0.31873326
Iteration 47, loss = 0.31763978
Iteration 48, loss = 0.31670896
Iteration 49, loss = 0.31572546
Iteration 50, loss = 0.31809186
Iteration 51, loss = 0.31705582
Iteration 52, loss = 0.31551140
Iteration 53, loss = 0.31647224
Iteration 54, loss = 0.31651529
Iteration 55, loss = 0.32073538
Iteration 56, loss = 0.32000289
Iteration 57, loss = 0.31562084
Iteration 58, loss = 0.31806983
Iteration 59, loss = 0.31765184
Iteration 60, loss = 0.31744877
Iteration 61, loss = 0.31950931
Iteration 62, loss = 0.31614750
Iteration 63, loss = 0.31713849
Training loss did not improve more than 

Iteration 40, loss = 0.33504902
Iteration 41, loss = 0.33976213
Iteration 42, loss = 0.33775816
Iteration 43, loss = 0.33373409
Iteration 44, loss = 0.33396311
Iteration 45, loss = 0.33397743
Iteration 46, loss = 0.33284187
Iteration 47, loss = 0.33287758
Iteration 48, loss = 0.33331762
Iteration 49, loss = 0.33288195
Iteration 50, loss = 0.33434581
Iteration 51, loss = 0.33152066
Iteration 52, loss = 0.33072777
Iteration 53, loss = 0.33244727
Iteration 54, loss = 0.33267717
Iteration 55, loss = 0.33242632
Iteration 56, loss = 0.33123841
Iteration 57, loss = 0.33063206
Iteration 58, loss = 0.33167071
Iteration 59, loss = 0.33211978
Iteration 60, loss = 0.33037530
Iteration 61, loss = 0.33127901
Iteration 62, loss = 0.33465506
Iteration 63, loss = 0.33421886
Iteration 64, loss = 0.33059255
Iteration 65, loss = 0.33070071
Iteration 66, loss = 0.33341699
Iteration 67, loss = 0.33201083
Iteration 68, loss = 0.33285971
Iteration 69, loss = 0.33332141
Iteration 70, loss = 0.33099213
Iteratio

Iteration 13, loss = 0.31094182
Iteration 14, loss = 0.30976370
Iteration 15, loss = 0.30846911
Iteration 16, loss = 0.30717196
Iteration 17, loss = 0.30624460
Iteration 18, loss = 0.30545589
Iteration 19, loss = 0.30335409
Iteration 20, loss = 0.30226938
Iteration 21, loss = 0.30358757
Iteration 22, loss = 0.30146392
Iteration 23, loss = 0.30137802
Iteration 24, loss = 0.30222618
Iteration 25, loss = 0.30281905
Iteration 26, loss = 0.30175045
Iteration 27, loss = 0.30136684
Iteration 28, loss = 0.30025938
Iteration 29, loss = 0.29983948
Iteration 30, loss = 0.30163811
Iteration 31, loss = 0.29832730
Iteration 32, loss = 0.30015293
Iteration 33, loss = 0.29926828
Iteration 34, loss = 0.29971102
Iteration 35, loss = 0.29649751
Iteration 36, loss = 0.29846865
Iteration 37, loss = 0.29734213
Iteration 38, loss = 0.29976627
Iteration 39, loss = 0.29677418
Iteration 40, loss = 0.29710461
Iteration 41, loss = 0.29836443
Iteration 42, loss = 0.29916090
Iteration 43, loss = 0.30149074
Iteratio

Iteration 71, loss = 0.32547386
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.81079845
Iteration 2, loss = 0.52549323
Iteration 3, loss = 0.44988076
Iteration 4, loss = 0.41397934
Iteration 5, loss = 0.39450466
Iteration 6, loss = 0.37931007
Iteration 7, loss = 0.36769415
Iteration 8, loss = 0.36167831
Iteration 9, loss = 0.35441476
Iteration 10, loss = 0.35105953
Iteration 11, loss = 0.34773227
Iteration 12, loss = 0.34531744
Iteration 13, loss = 0.34183327
Iteration 14, loss = 0.34130046
Iteration 15, loss = 0.34019806
Iteration 16, loss = 0.34125291
Iteration 17, loss = 0.34067873
Iteration 18, loss = 0.34025368
Iteration 19, loss = 0.33638571
Iteration 20, loss = 0.33820131
Iteration 21, loss = 0.33366034
Iteration 22, loss = 0.33575603
Iteration 23, loss = 0.33419370
Iteration 24, loss = 0.33143757
Iteration 25, loss = 0.33222377
Iteration 26, loss = 0.33419032
Iteration 27, loss = 0.33315572
Iteration 28, loss = 0.

Iteration 6, loss = 0.36983315
Iteration 7, loss = 0.35909821
Iteration 8, loss = 0.34986086
Iteration 9, loss = 0.34669064
Iteration 10, loss = 0.34200088
Iteration 11, loss = 0.33972450
Iteration 12, loss = 0.33852068
Iteration 13, loss = 0.33808703
Iteration 14, loss = 0.33331172
Iteration 15, loss = 0.33151116
Iteration 16, loss = 0.33190133
Iteration 17, loss = 0.33232826
Iteration 18, loss = 0.33142812
Iteration 19, loss = 0.32956445
Iteration 20, loss = 0.32883777
Iteration 21, loss = 0.33189868
Iteration 22, loss = 0.32784018
Iteration 23, loss = 0.32574047
Iteration 24, loss = 0.33078632
Iteration 25, loss = 0.32424526
Iteration 26, loss = 0.32436000
Iteration 27, loss = 0.32547871
Iteration 28, loss = 0.32814345
Iteration 29, loss = 0.32719808
Iteration 30, loss = 0.32571564
Iteration 31, loss = 0.33199231
Iteration 32, loss = 0.32724358
Iteration 33, loss = 0.32736487
Iteration 34, loss = 0.33469593
Iteration 35, loss = 0.32566768
Iteration 36, loss = 0.32193778
Iteration 37

Iteration 31, loss = 0.32912273
Iteration 32, loss = 0.32605346
Iteration 33, loss = 0.32743847
Iteration 34, loss = 0.32616330
Iteration 35, loss = 0.32782239
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.66451025
Iteration 2, loss = 0.47188155
Iteration 3, loss = 0.42625443
Iteration 4, loss = 0.39018348
Iteration 5, loss = 0.37245313
Iteration 6, loss = 0.36344899
Iteration 7, loss = 0.35372253
Iteration 8, loss = 0.34961717
Iteration 9, loss = 0.34755382
Iteration 10, loss = 0.34285834
Iteration 11, loss = 0.34126429
Iteration 12, loss = 0.34225978
Iteration 13, loss = 0.33900035
Iteration 14, loss = 0.33552581
Iteration 15, loss = 0.34239186
Iteration 16, loss = 0.33694483
Iteration 17, loss = 0.33842601
Iteration 18, loss = 0.33828304
Iteration 19, loss = 0.33852235
Iteration 20, loss = 0.33636103
Iteration 21, loss = 0.33675711
Iteration 22, loss = 0.33564078
Iteration 23, loss = 0.33429181
Iteration 24, loss = 0.

Iteration 44, loss = 0.31924158
Iteration 45, loss = 0.32703177
Iteration 46, loss = 0.32280612
Iteration 47, loss = 0.32194401
Iteration 48, loss = 0.32252226
Iteration 49, loss = 0.32173058
Iteration 50, loss = 0.32262778
Iteration 51, loss = 0.32675519
Iteration 52, loss = 0.32227716
Iteration 53, loss = 0.32065552
Iteration 54, loss = 0.31943478
Iteration 55, loss = 0.32054539
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.65789178
Iteration 2, loss = 0.45053227
Iteration 3, loss = 0.38912968
Iteration 4, loss = 0.35743805
Iteration 5, loss = 0.33774920
Iteration 6, loss = 0.32843661
Iteration 7, loss = 0.31925229
Iteration 8, loss = 0.31198471
Iteration 9, loss = 0.30807938
Iteration 10, loss = 0.31299508
Iteration 11, loss = 0.30818668
Iteration 12, loss = 0.30432868
Iteration 13, loss = 0.30285288
Iteration 14, loss = 0.30612753
Iteration 15, loss = 0.30126606
Iteration 16, loss = 0.30082176
Iteration 17, loss = 0.

Iteration 35, loss = 0.32517635
Iteration 36, loss = 0.32000467
Iteration 37, loss = 0.32206355
Iteration 38, loss = 0.31926622
Iteration 39, loss = 0.31832515
Iteration 40, loss = 0.32152074
Iteration 41, loss = 0.32649934
Iteration 42, loss = 0.32975831
Iteration 43, loss = 0.32200042
Iteration 44, loss = 0.32161544
Iteration 45, loss = 0.32095658
Iteration 46, loss = 0.31907183
Iteration 47, loss = 0.32043813
Iteration 48, loss = 0.32158526
Iteration 49, loss = 0.32112481
Iteration 50, loss = 0.32130663
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.75838576
Iteration 2, loss = 0.49711515
Iteration 3, loss = 0.43450385
Iteration 4, loss = 0.40491436
Iteration 5, loss = 0.38382317
Iteration 6, loss = 0.36967901
Iteration 7, loss = 0.36095374
Iteration 8, loss = 0.35273039
Iteration 9, loss = 0.35000578
Iteration 10, loss = 0.34671193
Iteration 11, loss = 0.34293634
Iteration 12, loss = 0.34295223
Iteration 13, loss = 0.

Iteration 29, loss = 0.32079069
Iteration 30, loss = 0.32758687
Iteration 31, loss = 0.32849974
Iteration 32, loss = 0.32844487
Iteration 33, loss = 0.32392583
Iteration 34, loss = 0.32363069
Iteration 35, loss = 0.32523204
Iteration 36, loss = 0.32112179
Iteration 37, loss = 0.32048175
Iteration 38, loss = 0.32151871
Iteration 39, loss = 0.32263635
Iteration 40, loss = 0.32116976
Iteration 41, loss = 0.32320395
Iteration 42, loss = 0.32287799
Iteration 43, loss = 0.32130777
Iteration 44, loss = 0.32453275
Iteration 45, loss = 0.32500999
Iteration 46, loss = 0.33200956
Iteration 47, loss = 0.33452016
Iteration 48, loss = 0.32428581
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.74416872
Iteration 2, loss = 0.47423874
Iteration 3, loss = 0.40603307
Iteration 4, loss = 0.37642463
Iteration 5, loss = 0.35275840
Iteration 6, loss = 0.33643605
Iteration 7, loss = 0.32898468
Iteration 8, loss = 0.31760488
Iteration 9, loss = 0.

Iteration 6, loss = 0.36643759
Iteration 7, loss = 0.35898875
Iteration 8, loss = 0.35239166
Iteration 9, loss = 0.34369591
Iteration 10, loss = 0.34913262
Iteration 11, loss = 0.34854486
Iteration 12, loss = 0.34322642
Iteration 13, loss = 0.33913970
Iteration 14, loss = 0.33892424
Iteration 15, loss = 0.33854764
Iteration 16, loss = 0.33524463
Iteration 17, loss = 0.33363049
Iteration 18, loss = 0.33535627
Iteration 19, loss = 0.33484788
Iteration 20, loss = 0.33185922
Iteration 21, loss = 0.33468571
Iteration 22, loss = 0.33037861
Iteration 23, loss = 0.33054138
Iteration 24, loss = 0.33110399
Iteration 25, loss = 0.32846972
Iteration 26, loss = 0.32929344
Iteration 27, loss = 0.33015988
Iteration 28, loss = 0.33154277
Iteration 29, loss = 0.33533661
Iteration 30, loss = 0.32766581
Iteration 31, loss = 0.33178640
Iteration 32, loss = 0.33342750
Iteration 33, loss = 0.33700782
Iteration 34, loss = 0.33864801
Iteration 35, loss = 0.32859609
Iteration 36, loss = 0.32728336
Iteration 37

Iteration 8, loss = 0.31536473
Iteration 9, loss = 0.31365486
Iteration 10, loss = 0.30771558
Iteration 11, loss = 0.30391375
Iteration 12, loss = 0.30656934
Iteration 13, loss = 0.30612970
Iteration 14, loss = 0.30353701
Iteration 15, loss = 0.30195778
Iteration 16, loss = 0.30569896
Iteration 17, loss = 0.30538579
Iteration 18, loss = 0.29783817
Iteration 19, loss = 0.30029452
Iteration 20, loss = 0.30379369
Iteration 21, loss = 0.29891623
Iteration 22, loss = 0.29631764
Iteration 23, loss = 0.29686496
Iteration 24, loss = 0.29959994
Iteration 25, loss = 0.29878529
Iteration 26, loss = 0.30176919
Iteration 27, loss = 0.29429538
Iteration 28, loss = 0.29319788
Iteration 29, loss = 0.29994750
Iteration 30, loss = 0.29671769
Iteration 31, loss = 0.29717762
Iteration 32, loss = 0.29161386
Iteration 33, loss = 0.29463368
Iteration 34, loss = 0.29257171
Iteration 35, loss = 0.29410177
Iteration 36, loss = 0.29593596
Iteration 37, loss = 0.29271270
Iteration 38, loss = 0.29234372
Iteration 

Iteration 49, loss = 0.32976653
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.75882830
Iteration 2, loss = 0.50139224
Iteration 3, loss = 0.42707725
Iteration 4, loss = 0.39614409
Iteration 5, loss = 0.37773560
Iteration 6, loss = 0.36360478
Iteration 7, loss = 0.36073590
Iteration 8, loss = 0.35240832
Iteration 9, loss = 0.35138939
Iteration 10, loss = 0.34996317
Iteration 11, loss = 0.34551890
Iteration 12, loss = 0.34847780
Iteration 13, loss = 0.34591264
Iteration 14, loss = 0.34472220
Iteration 15, loss = 0.34954254
Iteration 16, loss = 0.33788679
Iteration 17, loss = 0.34242032
Iteration 18, loss = 0.34480398
Iteration 19, loss = 0.34800972
Iteration 20, loss = 0.35173737
Iteration 21, loss = 0.34681291
Iteration 22, loss = 0.34837762
Iteration 23, loss = 0.34475331
Iteration 24, loss = 0.34694696
Iteration 25, loss = 0.34229842
Iteration 26, loss = 0.34141201
Iteration 27, loss = 0.34112873
Training loss did not i

Iteration 20, loss = 0.31438760
Iteration 21, loss = 0.31528171
Iteration 22, loss = 0.31622048
Iteration 23, loss = 0.32095919
Iteration 24, loss = 0.32067298
Iteration 25, loss = 0.32149924
Iteration 26, loss = 0.32595384
Iteration 27, loss = 0.31790610
Iteration 28, loss = 0.31831268
Iteration 29, loss = 0.31310123
Iteration 30, loss = 0.31394945
Iteration 31, loss = 0.31774658
Iteration 32, loss = 0.31712356
Iteration 33, loss = 0.31528965
Iteration 34, loss = 0.31875469
Iteration 35, loss = 0.31536020
Iteration 36, loss = 0.31866741
Iteration 37, loss = 0.31629671
Iteration 38, loss = 0.32035772
Iteration 39, loss = 0.32014385
Iteration 40, loss = 0.32167751
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.67548106
Iteration 2, loss = 0.47195590
Iteration 3, loss = 0.41042986
Iteration 4, loss = 0.37639835
Iteration 5, loss = 0.35417400
Iteration 6, loss = 0.34330449
Iteration 7, loss = 0.33455371
Iteration 8, loss = 0

Iteration 32, loss = 0.33315536
Iteration 33, loss = 0.33594937
Iteration 34, loss = 0.32846454
Iteration 35, loss = 0.33313521
Iteration 36, loss = 0.33665347
Iteration 37, loss = 0.33409327
Iteration 38, loss = 0.33192349
Iteration 39, loss = 0.33307184
Iteration 40, loss = 0.33004992
Iteration 41, loss = 0.33499949
Iteration 42, loss = 0.33015163
Iteration 43, loss = 0.33324152
Iteration 44, loss = 0.33709651
Iteration 45, loss = 0.33656122
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.68436270
Iteration 2, loss = 0.47365428
Iteration 3, loss = 0.41454850
Iteration 4, loss = 0.38126876
Iteration 5, loss = 0.36352220
Iteration 6, loss = 0.35158810
Iteration 7, loss = 0.34222382
Iteration 8, loss = 0.33827698
Iteration 9, loss = 0.33312297
Iteration 10, loss = 0.33204142
Iteration 11, loss = 0.32724890
Iteration 12, loss = 0.32843430
Iteration 13, loss = 0.32984877
Iteration 14, loss = 0.33215402
Iteration 15, loss = 0.

Iteration 8, loss = 0.33310854
Iteration 9, loss = 0.34161648
Iteration 10, loss = 0.33145826
Iteration 11, loss = 0.33428913
Iteration 12, loss = 0.33073524
Iteration 13, loss = 0.33599259
Iteration 14, loss = 0.32718277
Iteration 15, loss = 0.32460031
Iteration 16, loss = 0.32496973
Iteration 17, loss = 0.32148852
Iteration 18, loss = 0.32106376
Iteration 19, loss = 0.32721584
Iteration 20, loss = 0.32797979
Iteration 21, loss = 0.33056054
Iteration 22, loss = 0.32473249
Iteration 23, loss = 0.32552100
Iteration 24, loss = 0.32702631
Iteration 25, loss = 0.32565155
Iteration 26, loss = 0.32542377
Iteration 27, loss = 0.32098789
Iteration 28, loss = 0.32557644
Iteration 29, loss = 0.31937393
Iteration 30, loss = 0.32342251
Iteration 31, loss = 0.32316797
Iteration 32, loss = 0.33120435
Iteration 33, loss = 0.32543441
Iteration 34, loss = 0.32228435
Iteration 35, loss = 0.32092048
Iteration 36, loss = 0.32569357
Iteration 37, loss = 0.32031470
Iteration 38, loss = 0.32687220
Iteration 

Iteration 23, loss = 0.30128183
Iteration 24, loss = 0.29740009
Iteration 25, loss = 0.29917591
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.62653625
Iteration 2, loss = 0.43556521
Iteration 3, loss = 0.38499172
Iteration 4, loss = 0.35776970
Iteration 5, loss = 0.33869565
Iteration 6, loss = 0.33858975
Iteration 7, loss = 0.32988594
Iteration 8, loss = 0.32918352
Iteration 9, loss = 0.32767770
Iteration 10, loss = 0.32559516
Iteration 11, loss = 0.32740135
Iteration 12, loss = 0.32827011
Iteration 13, loss = 0.32327384
Iteration 14, loss = 0.32158811
Iteration 15, loss = 0.31771860
Iteration 16, loss = 0.32243494
Iteration 17, loss = 0.32480625
Iteration 18, loss = 0.33232995
Iteration 19, loss = 0.32674289
Iteration 20, loss = 0.33275374
Iteration 21, loss = 0.31981108
Iteration 22, loss = 0.31871645
Iteration 23, loss = 0.32262438
Iteration 24, loss = 0.31624108
Iteration 25, loss = 0.31655122
Iteration 26, loss = 0.

Iteration 18, loss = 0.32554309
Iteration 19, loss = 0.33392638
Iteration 20, loss = 0.32810916
Iteration 21, loss = 0.32902321
Iteration 22, loss = 0.32540856
Iteration 23, loss = 0.32511634
Iteration 24, loss = 0.32904306
Iteration 25, loss = 0.31886574
Iteration 26, loss = 0.32197626
Iteration 27, loss = 0.32377689
Iteration 28, loss = 0.32422366
Iteration 29, loss = 0.33952758
Iteration 30, loss = 0.32364623
Iteration 31, loss = 0.32242824
Iteration 32, loss = 0.32172105
Iteration 33, loss = 0.32319421
Iteration 34, loss = 0.31787157
Iteration 35, loss = 0.31802859
Iteration 36, loss = 0.32306542
Iteration 37, loss = 0.32376231
Iteration 38, loss = 0.31942103
Iteration 39, loss = 0.31959729
Iteration 40, loss = 0.32062604
Iteration 41, loss = 0.32382538
Iteration 42, loss = 0.32550849
Iteration 43, loss = 0.32157715
Iteration 44, loss = 0.32382938
Iteration 45, loss = 0.33655613
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, l

Iteration 27, loss = 0.32776289
Iteration 28, loss = 0.32959534
Iteration 29, loss = 0.33219376
Iteration 30, loss = 0.32367372
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.63162114
Iteration 2, loss = 0.46202528
Iteration 3, loss = 0.40880454
Iteration 4, loss = 0.38183514
Iteration 5, loss = 0.36325745
Iteration 6, loss = 0.35062635
Iteration 7, loss = 0.35733444
Iteration 8, loss = 0.34199291
Iteration 9, loss = 0.33854971
Iteration 10, loss = 0.33590480
Iteration 11, loss = 0.33517752
Iteration 12, loss = 0.33761641
Iteration 13, loss = 0.34610988
Iteration 14, loss = 0.34110372
Iteration 15, loss = 0.33588483
Iteration 16, loss = 0.33239450
Iteration 17, loss = 0.33237353
Iteration 18, loss = 0.33127092
Iteration 19, loss = 0.32829897
Iteration 20, loss = 0.33821705
Iteration 21, loss = 0.33398007
Iteration 22, loss = 0.33568698
Iteration 23, loss = 0.34505847
Iteration 24, loss = 0.33601095
Iteration 25, loss = 0.

Iteration 11, loss = 0.30253384
Iteration 12, loss = 0.29875588
Iteration 13, loss = 0.30017125
Iteration 14, loss = 0.30135407
Iteration 15, loss = 0.30639685
Iteration 16, loss = 0.30147136
Iteration 17, loss = 0.29815106
Iteration 18, loss = 0.30448828
Iteration 19, loss = 0.30665020
Iteration 20, loss = 0.30360977
Iteration 21, loss = 0.30245776
Iteration 22, loss = 0.30656559
Iteration 23, loss = 0.29947421
Iteration 24, loss = 0.29652871
Iteration 25, loss = 0.30038664
Iteration 26, loss = 0.29921073
Iteration 27, loss = 0.30418077
Iteration 28, loss = 0.29344975
Iteration 29, loss = 0.29890269
Iteration 30, loss = 0.29815705
Iteration 31, loss = 0.29380841
Iteration 32, loss = 0.29191471
Iteration 33, loss = 0.29436725
Iteration 34, loss = 0.29384019
Iteration 35, loss = 0.29305780
Iteration 36, loss = 0.29455252
Iteration 37, loss = 0.29433508
Iteration 38, loss = 0.29855421
Iteration 39, loss = 0.29347889
Iteration 40, loss = 0.29825967
Iteration 41, loss = 0.29463863
Iteratio

Iteration 13, loss = 0.34351277
Iteration 14, loss = 0.35111544
Iteration 15, loss = 0.34493119
Iteration 16, loss = 0.34632119
Iteration 17, loss = 0.34661293
Iteration 18, loss = 0.33801649
Iteration 19, loss = 0.34003464
Iteration 20, loss = 0.34074767
Iteration 21, loss = 0.33691910
Iteration 22, loss = 0.33872318
Iteration 23, loss = 0.34255247
Iteration 24, loss = 0.34085369
Iteration 25, loss = 0.34611432
Iteration 26, loss = 0.34192013
Iteration 27, loss = 0.34072907
Iteration 28, loss = 0.33770030
Iteration 29, loss = 0.34880903
Iteration 30, loss = 0.34573880
Iteration 31, loss = 0.33992362
Iteration 32, loss = 0.34545448
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.66083374
Iteration 2, loss = 0.45118093
Iteration 3, loss = 0.39060800
Iteration 4, loss = 0.36538852
Iteration 5, loss = 0.34997977
Iteration 6, loss = 0.34180380
Iteration 7, loss = 0.33560866
Iteration 8, loss = 0.33339398
Iteration 9, loss = 0.

Iteration 4, loss = 0.36224169
Iteration 5, loss = 0.35221755
Iteration 6, loss = 0.33430528
Iteration 7, loss = 0.33918423
Iteration 8, loss = 0.33278284
Iteration 9, loss = 0.33286192
Iteration 10, loss = 0.32855673
Iteration 11, loss = 0.32153892
Iteration 12, loss = 0.32147598
Iteration 13, loss = 0.32139894
Iteration 14, loss = 0.32539602
Iteration 15, loss = 0.32463461
Iteration 16, loss = 0.32850392
Iteration 17, loss = 0.31985545
Iteration 18, loss = 0.31836954
Iteration 19, loss = 0.31894221
Iteration 20, loss = 0.32319261
Iteration 21, loss = 0.32469622
Iteration 22, loss = 0.33655488
Iteration 23, loss = 0.32610005
Iteration 24, loss = 0.32916437
Iteration 25, loss = 0.32516645
Iteration 26, loss = 0.33231411
Iteration 27, loss = 0.32534523
Iteration 28, loss = 0.31895267
Iteration 29, loss = 0.31717194
Iteration 30, loss = 0.31711561
Iteration 31, loss = 0.31602335
Iteration 32, loss = 0.32210639
Iteration 33, loss = 0.32184010
Iteration 34, loss = 0.32455024
Iteration 35, 

Iteration 33, loss = 0.34005847
Iteration 34, loss = 0.33934679
Iteration 35, loss = 0.33846900
Iteration 36, loss = 0.33754217
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.67286984
Iteration 2, loss = 0.43750359
Iteration 3, loss = 0.39617261
Iteration 4, loss = 0.36646627
Iteration 5, loss = 0.35360020
Iteration 6, loss = 0.34304560
Iteration 7, loss = 0.34497716
Iteration 8, loss = 0.33458615
Iteration 9, loss = 0.33250739
Iteration 10, loss = 0.34091938
Iteration 11, loss = 0.34895231
Iteration 12, loss = 0.33674831
Iteration 13, loss = 0.33350088
Iteration 14, loss = 0.32924557
Iteration 15, loss = 0.32433687
Iteration 16, loss = 0.32964719
Iteration 17, loss = 0.32357706
Iteration 18, loss = 0.32874962
Iteration 19, loss = 0.33120540
Iteration 20, loss = 0.33354984
Iteration 21, loss = 0.33120343
Iteration 22, loss = 0.33254313
Iteration 23, loss = 0.32319074
Iteration 24, loss = 0.32516860
Iteration 25, loss = 0.

Iteration 29, loss = 0.33022817
Iteration 30, loss = 0.33270414
Iteration 31, loss = 0.33783939
Iteration 32, loss = 0.33442240
Iteration 33, loss = 0.33287898
Iteration 34, loss = 0.33950485
Iteration 35, loss = 0.33847963
Iteration 36, loss = 0.33624218
Iteration 37, loss = 0.33353693
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.65548961
Iteration 2, loss = 0.44438420
Iteration 3, loss = 0.40111819
Iteration 4, loss = 0.37623390
Iteration 5, loss = 0.36567905
Iteration 6, loss = 0.35413429
Iteration 7, loss = 0.34881631
Iteration 8, loss = 0.34942046
Iteration 9, loss = 0.34848372
Iteration 10, loss = 0.34631300
Iteration 11, loss = 0.34827721
Iteration 12, loss = 0.34526584
Iteration 13, loss = 0.33884425
Iteration 14, loss = 0.34437840
Iteration 15, loss = 0.35220072
Iteration 16, loss = 0.35399917
Iteration 17, loss = 0.35498202
Iteration 18, loss = 0.34254110
Iteration 19, loss = 0.34434164
Iteration 20, loss = 0.

Iteration 12, loss = 0.32581027
Iteration 13, loss = 0.32365383
Iteration 14, loss = 0.32316029
Iteration 15, loss = 0.32005981
Iteration 16, loss = 0.32523376
Iteration 17, loss = 0.31834793
Iteration 18, loss = 0.32879520
Iteration 19, loss = 0.31597104
Iteration 20, loss = 0.31916209
Iteration 21, loss = 0.32254858
Iteration 22, loss = 0.32646802
Iteration 23, loss = 0.32603085
Iteration 24, loss = 0.32097281
Iteration 25, loss = 0.31660843
Iteration 26, loss = 0.31330264
Iteration 27, loss = 0.31573561
Iteration 28, loss = 0.31503945
Iteration 29, loss = 0.31413830
Iteration 30, loss = 0.32512056
Iteration 31, loss = 0.32444655
Iteration 32, loss = 0.32298575
Iteration 33, loss = 0.32279518
Iteration 34, loss = 0.33789503
Iteration 35, loss = 0.32433441
Iteration 36, loss = 0.32954991
Iteration 37, loss = 0.33316752
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.61193218
Iteration 2, loss = 0.42103259
Iteration 3, los

Iteration 20, loss = 0.34164121
Iteration 21, loss = 0.35065779
Iteration 22, loss = 0.34655153
Iteration 23, loss = 0.35247987
Iteration 24, loss = 0.34237366
Iteration 25, loss = 0.33399458
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.61334255
Iteration 2, loss = 0.42162372
Iteration 3, loss = 0.38265328
Iteration 4, loss = 0.35481465
Iteration 5, loss = 0.34269768
Iteration 6, loss = 0.34160423
Iteration 7, loss = 0.34299244
Iteration 8, loss = 0.33698781
Iteration 9, loss = 0.33422421
Iteration 10, loss = 0.33001895
Iteration 11, loss = 0.33850386
Iteration 12, loss = 0.33359059
Iteration 13, loss = 0.32997391
Iteration 14, loss = 0.33069319
Iteration 15, loss = 0.32918034
Iteration 16, loss = 0.33285141
Iteration 17, loss = 0.33148873
Iteration 18, loss = 0.33217834
Iteration 19, loss = 0.33550407
Iteration 20, loss = 0.33499035
Iteration 21, loss = 0.33630932
Iteration 22, loss = 0.35107404
Iteration 23, loss = 0.

Iteration 55, loss = 0.34840236
Iteration 56, loss = 0.32953131
Iteration 57, loss = 0.33305511
Iteration 58, loss = 0.32079913
Iteration 59, loss = 0.33372432
Iteration 60, loss = 0.32778864
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.69244577
Iteration 2, loss = 0.45249751
Iteration 3, loss = 0.39841237
Iteration 4, loss = 0.38168103
Iteration 5, loss = 0.36046556
Iteration 6, loss = 0.34939969
Iteration 7, loss = 0.34241369
Iteration 8, loss = 0.34096325
Iteration 9, loss = 0.33983339
Iteration 10, loss = 0.34063335
Iteration 11, loss = 0.33765988
Iteration 12, loss = 0.34446438
Iteration 13, loss = 0.35069111
Iteration 14, loss = 0.34659206
Iteration 15, loss = 0.33968557
Iteration 16, loss = 0.33172973
Iteration 17, loss = 0.33544030
Iteration 18, loss = 0.33175774
Iteration 19, loss = 0.33548667
Iteration 20, loss = 0.33369903
Iteration 21, loss = 0.33425936
Iteration 22, loss = 0.33837692
Iteration 23, loss = 0.

Iteration 9, loss = 0.32965915
Iteration 10, loss = 0.33743539
Iteration 11, loss = 0.32480087
Iteration 12, loss = 0.32705424
Iteration 13, loss = 0.32623754
Iteration 14, loss = 0.32085934
Iteration 15, loss = 0.33578250
Iteration 16, loss = 0.32764583
Iteration 17, loss = 0.32044193
Iteration 18, loss = 0.32242935
Iteration 19, loss = 0.32249188
Iteration 20, loss = 0.31924202
Iteration 21, loss = 0.32333483
Iteration 22, loss = 0.32727759
Iteration 23, loss = 0.32348795
Iteration 24, loss = 0.32212965
Iteration 25, loss = 0.32104729
Iteration 26, loss = 0.32393617
Iteration 27, loss = 0.31618847
Iteration 28, loss = 0.31567776
Iteration 29, loss = 0.31858870
Iteration 30, loss = 0.31531270
Iteration 31, loss = 0.31907121
Iteration 32, loss = 0.31678658
Iteration 33, loss = 0.31870159
Iteration 34, loss = 0.31615732
Iteration 35, loss = 0.31896465
Iteration 36, loss = 0.32172443
Iteration 37, loss = 0.32132789
Iteration 38, loss = 0.31364947
Iteration 39, loss = 0.31710694
Iteration

Iteration 21, loss = 0.34621980
Iteration 22, loss = 0.34386229
Iteration 23, loss = 0.34026403
Iteration 24, loss = 0.33711725
Iteration 25, loss = 0.33418156
Iteration 26, loss = 0.35499491
Iteration 27, loss = 0.34612860
Iteration 28, loss = 0.34021481
Iteration 29, loss = 0.33855410
Iteration 30, loss = 0.33785564
Iteration 31, loss = 0.33457683
Iteration 32, loss = 0.33548888
Iteration 33, loss = 0.33253326
Iteration 34, loss = 0.33823842
Iteration 35, loss = 0.33137161
Iteration 36, loss = 0.33070118
Iteration 37, loss = 0.34129539
Iteration 38, loss = 0.33599193
Iteration 39, loss = 0.33464643
Iteration 40, loss = 0.33451600
Iteration 41, loss = 0.33671283
Iteration 42, loss = 0.33492972
Iteration 43, loss = 0.33427465
Iteration 44, loss = 0.34143497
Iteration 45, loss = 0.34916038
Iteration 46, loss = 0.34601199
Iteration 47, loss = 0.33712172
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.64625750
Iteration 2, lo

Iteration 21, loss = 0.33301086
Iteration 22, loss = 0.32833592
Iteration 23, loss = 0.35260855
Iteration 24, loss = 0.34925916
Iteration 25, loss = 0.33649107
Iteration 26, loss = 0.33179518
Iteration 27, loss = 0.33221459
Iteration 28, loss = 0.32634006
Iteration 29, loss = 0.32917626
Iteration 30, loss = 0.33175677
Iteration 31, loss = 0.32599531
Iteration 32, loss = 0.32335600
Iteration 33, loss = 0.32179534
Iteration 34, loss = 0.32416868
Iteration 35, loss = 0.32385598
Iteration 36, loss = 0.32441456
Iteration 37, loss = 0.32542378
Iteration 38, loss = 0.32143084
Iteration 39, loss = 0.32489029
Iteration 40, loss = 0.32178130
Iteration 41, loss = 0.32289695
Iteration 42, loss = 0.33203997
Iteration 43, loss = 0.32906908
Iteration 44, loss = 0.33268748
Iteration 45, loss = 0.32681865
Iteration 46, loss = 0.33328180
Iteration 47, loss = 0.33003866
Iteration 48, loss = 0.32872308
Iteration 49, loss = 0.32530973
Training loss did not improve more than tol=0.000100 for 10 consecutive 

Iteration 26, loss = 0.34734091
Iteration 27, loss = 0.33823714
Iteration 28, loss = 0.33230057
Iteration 29, loss = 0.32442511
Iteration 30, loss = 0.32979524
Iteration 31, loss = 0.32285565
Iteration 32, loss = 0.32839583
Iteration 33, loss = 0.32092000
Iteration 34, loss = 0.33500706
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.65424589
Iteration 2, loss = 0.44031940
Iteration 3, loss = 0.38136978
Iteration 4, loss = 0.35728806
Iteration 5, loss = 0.34052344
Iteration 6, loss = 0.33132626
Iteration 7, loss = 0.33228935
Iteration 8, loss = 0.32896070
Iteration 9, loss = 0.32995707
Iteration 10, loss = 0.33083497
Iteration 11, loss = 0.33302421
Iteration 12, loss = 0.33382008
Iteration 13, loss = 0.32991556
Iteration 14, loss = 0.32836481
Iteration 15, loss = 0.32843098
Iteration 16, loss = 0.32678080
Iteration 17, loss = 0.32402282
Iteration 18, loss = 0.32650291
Iteration 19, loss = 0.32430530
Iteration 20, loss = 0.

Iteration 13, loss = 0.34282826
Iteration 14, loss = 0.34688978
Iteration 15, loss = 0.34443954
Iteration 16, loss = 0.34532943
Iteration 17, loss = 0.35602273
Iteration 18, loss = 0.36165713
Iteration 19, loss = 0.34131536
Iteration 20, loss = 0.34449660
Iteration 21, loss = 0.34184507
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.69207456
Iteration 2, loss = 0.47842376
Iteration 3, loss = 0.41185952
Iteration 4, loss = 0.37665680
Iteration 5, loss = 0.35418984
Iteration 6, loss = 0.34641066
Iteration 7, loss = 0.34017787
Iteration 8, loss = 0.33355057
Iteration 9, loss = 0.33960413
Iteration 10, loss = 0.33096470
Iteration 11, loss = 0.33194147
Iteration 12, loss = 0.33060659
Iteration 13, loss = 0.33422713
Iteration 14, loss = 0.34392713
Iteration 15, loss = 0.33300444
Iteration 16, loss = 0.33386520
Iteration 17, loss = 0.33759910
Iteration 18, loss = 0.33621480
Iteration 19, loss = 0.32713575
Iteration 20, loss = 0.

Iteration 32, loss = 0.34341410
Iteration 33, loss = 0.35417316
Iteration 34, loss = 0.34438760
Iteration 35, loss = 0.34270376
Iteration 36, loss = 0.36046022
Iteration 37, loss = 0.34298921
Iteration 38, loss = 0.34198699
Iteration 39, loss = 0.34833690
Iteration 40, loss = 0.33989319
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.63185357
Iteration 2, loss = 0.45747642
Iteration 3, loss = 0.40386223
Iteration 4, loss = 0.37526025
Iteration 5, loss = 0.37924073
Iteration 6, loss = 0.36233434
Iteration 7, loss = 0.36107608
Iteration 8, loss = 0.35629316
Iteration 9, loss = 0.35312324
Iteration 10, loss = 0.35445883
Iteration 11, loss = 0.35506066
Iteration 12, loss = 0.34548855
Iteration 13, loss = 0.34247118
Iteration 14, loss = 0.34961595
Iteration 15, loss = 0.35183974
Iteration 16, loss = 0.35903766
Iteration 17, loss = 0.35598465
Iteration 18, loss = 0.35781744
Iteration 19, loss = 0.36720687
Iteration 20, loss = 0.

Iteration 28, loss = 0.33330055
Iteration 29, loss = 0.32447302
Iteration 30, loss = 0.32185585
Iteration 31, loss = 0.32317485
Iteration 32, loss = 0.32691311
Iteration 33, loss = 0.33412045
Iteration 34, loss = 0.32867592
Iteration 35, loss = 0.32445278
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.74049549
Iteration 2, loss = 0.48105233
Iteration 3, loss = 0.40706115
Iteration 4, loss = 0.37384597
Iteration 5, loss = 0.35969573
Iteration 6, loss = 0.34146918
Iteration 7, loss = 0.33813556
Iteration 8, loss = 0.33380469
Iteration 9, loss = 0.34097932
Iteration 10, loss = 0.33985827
Iteration 11, loss = 0.33082243
Iteration 12, loss = 0.33093510
Iteration 13, loss = 0.33337657
Iteration 14, loss = 0.32823548
Iteration 15, loss = 0.33105657
Iteration 16, loss = 0.34681703
Iteration 17, loss = 0.33734883
Iteration 18, loss = 0.33114125
Iteration 19, loss = 0.34502549
Iteration 20, loss = 0.33747415
Iteration 21, loss = 0.

Iteration 1, loss = 0.61839481
Iteration 2, loss = 0.41817068
Iteration 3, loss = 0.37537666
Iteration 4, loss = 0.35372293
Iteration 5, loss = 0.34095700
Iteration 6, loss = 0.33257357
Iteration 7, loss = 0.33099324
Iteration 8, loss = 0.32798605
Iteration 9, loss = 0.32896562
Iteration 10, loss = 0.32787900
Iteration 11, loss = 0.33039359
Iteration 12, loss = 0.33389473
Iteration 13, loss = 0.32848997
Iteration 14, loss = 0.33500558
Iteration 15, loss = 0.33529787
Iteration 16, loss = 0.33585117
Iteration 17, loss = 0.33705977
Iteration 18, loss = 0.32685418
Iteration 19, loss = 0.31633030
Iteration 20, loss = 0.32258859
Iteration 21, loss = 0.32160994
Iteration 22, loss = 0.32092450
Iteration 23, loss = 0.32179194
Iteration 24, loss = 0.32585861
Iteration 25, loss = 0.32745651
Iteration 26, loss = 0.31875726
Iteration 27, loss = 0.32255323
Iteration 28, loss = 0.32306630
Iteration 29, loss = 0.32144666
Iteration 30, loss = 0.32612849
Training loss did not improve more than tol=0.000

Iteration 13, loss = 0.32799663
Iteration 14, loss = 0.33654303
Iteration 15, loss = 0.35639627
Iteration 16, loss = 0.35255298
Iteration 17, loss = 0.33730042
Iteration 18, loss = 0.33284332
Iteration 19, loss = 0.34548891
Iteration 20, loss = 0.34690056
Iteration 21, loss = 0.33677362
Iteration 22, loss = 0.32443998
Iteration 23, loss = 0.32933861
Iteration 24, loss = 0.33799525
Iteration 25, loss = 0.33268326
Iteration 26, loss = 0.33771664
Iteration 27, loss = 0.33282702
Iteration 28, loss = 0.33041367
Iteration 29, loss = 0.33049551
Iteration 30, loss = 0.33365083
Iteration 31, loss = 0.33289982
Iteration 32, loss = 0.32266409
Iteration 33, loss = 0.32928843
Iteration 34, loss = 0.34420735
Iteration 35, loss = 0.35998078
Iteration 36, loss = 0.35574479
Iteration 37, loss = 0.33195156
Iteration 38, loss = 0.33211341
Iteration 39, loss = 0.32365417
Iteration 40, loss = 0.32776189
Iteration 41, loss = 0.32313276
Iteration 42, loss = 0.32878731
Iteration 43, loss = 0.32606973
Training

Iteration 12, loss = 0.33326671
Iteration 13, loss = 0.32741768
Iteration 14, loss = 0.32670912
Iteration 15, loss = 0.32783929
Iteration 16, loss = 0.33679458
Iteration 17, loss = 0.33733176
Iteration 18, loss = 0.34729507
Iteration 19, loss = 0.33475279
Iteration 20, loss = 0.33241208
Iteration 21, loss = 0.33717725
Iteration 22, loss = 0.33149046
Iteration 23, loss = 0.33020935
Iteration 24, loss = 0.32978904
Iteration 25, loss = 0.32834954
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.65278500
Iteration 2, loss = 0.44829528
Iteration 3, loss = 0.40027772
Iteration 4, loss = 0.36940986
Iteration 5, loss = 0.35456878
Iteration 6, loss = 0.34928486
Iteration 7, loss = 0.34398038
Iteration 8, loss = 0.35470585
Iteration 9, loss = 0.34888747
Iteration 10, loss = 0.34232872
Iteration 11, loss = 0.34923437
Iteration 12, loss = 0.34028998
Iteration 13, loss = 0.34173213
Iteration 14, loss = 0.33857685
Iteration 15, loss = 0.

Iteration 13, loss = 0.33266313
Iteration 14, loss = 0.32631021
Iteration 15, loss = 0.32103284
Iteration 16, loss = 0.32647811
Iteration 17, loss = 0.32770197
Iteration 18, loss = 0.32354004
Iteration 19, loss = 0.32226789
Iteration 20, loss = 0.32040572
Iteration 21, loss = 0.32683454
Iteration 22, loss = 0.33527179
Iteration 23, loss = 0.31783607
Iteration 24, loss = 0.32472500
Iteration 25, loss = 0.31985789
Iteration 26, loss = 0.32252171
Iteration 27, loss = 0.32850908
Iteration 28, loss = 0.34185335
Iteration 29, loss = 0.32958464
Iteration 30, loss = 0.32196369
Iteration 31, loss = 0.32643176
Iteration 32, loss = 0.32239645
Iteration 33, loss = 0.32216104
Iteration 34, loss = 0.32898815
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.60551377
Iteration 2, loss = 0.41594406
Iteration 3, loss = 0.37234570
Iteration 4, loss = 0.34811188
Iteration 5, loss = 0.34289468
Iteration 6, loss = 0.34643334
Iteration 7, loss = 

Iteration 21, loss = 0.33261768
Iteration 22, loss = 0.34599247
Iteration 23, loss = 0.33433187
Iteration 24, loss = 0.32671741
Iteration 25, loss = 0.33700586
Iteration 26, loss = 0.33303672
Iteration 27, loss = 0.33469808
Iteration 28, loss = 0.33475896
Iteration 29, loss = 0.32493845
Iteration 30, loss = 0.33702287
Iteration 31, loss = 0.32852647
Iteration 32, loss = 0.32857289
Iteration 33, loss = 0.33097952
Iteration 34, loss = 0.33500391
Iteration 35, loss = 0.33602380
Iteration 36, loss = 0.33172995
Iteration 37, loss = 0.33191741
Iteration 38, loss = 0.33758298
Iteration 39, loss = 0.33668350
Iteration 40, loss = 0.32538747
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.58029002
Iteration 2, loss = 0.39886566
Iteration 3, loss = 0.35476913
Iteration 4, loss = 0.33378299
Iteration 5, loss = 0.31189847
Iteration 6, loss = 0.30520135
Iteration 7, loss = 0.30440047
Iteration 8, loss = 0.30366997
Iteration 9, loss = 0.

Iteration 27, loss = 0.34710596
Iteration 28, loss = 0.34643030
Iteration 29, loss = 0.34862497
Iteration 30, loss = 0.36021215
Iteration 31, loss = 0.34471424
Iteration 32, loss = 0.34321659
Iteration 33, loss = 0.34012618
Iteration 34, loss = 0.34289391
Iteration 35, loss = 0.35563582
Iteration 36, loss = 0.34498095
Iteration 37, loss = 0.35006922
Iteration 38, loss = 0.36041344
Iteration 39, loss = 0.36058281
Iteration 40, loss = 0.36656106
Iteration 41, loss = 0.33862555
Iteration 42, loss = 0.33540737
Iteration 43, loss = 0.34201315
Iteration 44, loss = 0.34404333
Iteration 45, loss = 0.34978995
Iteration 46, loss = 0.34408447
Iteration 47, loss = 0.33995395
Iteration 48, loss = 0.33929487
Iteration 49, loss = 0.33932153
Iteration 50, loss = 0.34279362
Iteration 51, loss = 0.33950011
Iteration 52, loss = 0.33855103
Iteration 53, loss = 0.33924897
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.71322654
Iteration 2, lo

Iteration 42, loss = 0.31875280
Iteration 43, loss = 0.31800766
Iteration 44, loss = 0.31643257
Iteration 45, loss = 0.32759699
Iteration 46, loss = 0.33676451
Iteration 47, loss = 0.32589602
Iteration 48, loss = 0.32470367
Iteration 49, loss = 0.33117821
Iteration 50, loss = 0.33775862
Iteration 51, loss = 0.33791347
Iteration 52, loss = 0.33996864
Iteration 53, loss = 0.33992578
Iteration 54, loss = 0.33485354
Iteration 55, loss = 0.31866644
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.70159661
Iteration 2, loss = 0.44961454
Iteration 3, loss = 0.39564192
Iteration 4, loss = 0.35993391
Iteration 5, loss = 0.34793234
Iteration 6, loss = 0.34144834
Iteration 7, loss = 0.35182811
Iteration 8, loss = 0.34813466
Iteration 9, loss = 0.35234476
Iteration 10, loss = 0.35317806
Iteration 11, loss = 0.35588404
Iteration 12, loss = 0.34502002
Iteration 13, loss = 0.33589694
Iteration 14, loss = 0.33005229
Iteration 15, loss = 0.

Iteration 33, loss = 0.31123235
Iteration 34, loss = 0.30985170
Iteration 35, loss = 0.31530404
Iteration 36, loss = 0.30689934
Iteration 37, loss = 0.30151774
Iteration 38, loss = 0.29725425
Iteration 39, loss = 0.30304693
Iteration 40, loss = 0.30599152
Iteration 41, loss = 0.31037568
Iteration 42, loss = 0.32810981
Iteration 43, loss = 0.33926032
Iteration 44, loss = 0.32458239
Iteration 45, loss = 0.30129088
Iteration 46, loss = 0.30540540
Iteration 47, loss = 0.30797677
Iteration 48, loss = 0.30250432
Iteration 49, loss = 0.30050581
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.65634834
Iteration 2, loss = 0.43089743
Iteration 3, loss = 0.37968784
Iteration 4, loss = 0.35230423
Iteration 5, loss = 0.34107334
Iteration 6, loss = 0.33238265
Iteration 7, loss = 0.33753497
Iteration 8, loss = 0.34785558
Iteration 9, loss = 0.34759751
Iteration 10, loss = 0.34164367
Iteration 11, loss = 0.32997694
Iteration 12, loss = 0.

Iteration 5, loss = 0.34741124
Iteration 6, loss = 0.33694727
Iteration 7, loss = 0.34140632
Iteration 8, loss = 0.33772340
Iteration 9, loss = 0.33090832
Iteration 10, loss = 0.33635783
Iteration 11, loss = 0.34038734
Iteration 12, loss = 0.33460745
Iteration 13, loss = 0.33791137
Iteration 14, loss = 0.33884542
Iteration 15, loss = 0.33235589
Iteration 16, loss = 0.33145046
Iteration 17, loss = 0.32990644
Iteration 18, loss = 0.33365498
Iteration 19, loss = 0.33383529
Iteration 20, loss = 0.33382252
Iteration 21, loss = 0.34109768
Iteration 22, loss = 0.35135159
Iteration 23, loss = 0.33298683
Iteration 24, loss = 0.33074302
Iteration 25, loss = 0.33062124
Iteration 26, loss = 0.32794047
Iteration 27, loss = 0.32763006
Iteration 28, loss = 0.33313938
Iteration 29, loss = 0.32927320
Iteration 30, loss = 0.33835386
Iteration 31, loss = 0.33259448
Iteration 32, loss = 0.32492328
Iteration 33, loss = 0.33408710
Iteration 34, loss = 0.33455741
Iteration 35, loss = 0.32750625
Iteration 36,

Iteration 21, loss = 0.36129660
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.66311885
Iteration 2, loss = 0.44173833
Iteration 3, loss = 0.39872769
Iteration 4, loss = 0.38031984
Iteration 5, loss = 0.36944271
Iteration 6, loss = 0.35443324
Iteration 7, loss = 0.35405449
Iteration 8, loss = 0.35568928
Iteration 9, loss = 0.36209618
Iteration 10, loss = 0.35127679
Iteration 11, loss = 0.34755479
Iteration 12, loss = 0.35556393
Iteration 13, loss = 0.34373756
Iteration 14, loss = 0.34883568
Iteration 15, loss = 0.34749654
Iteration 16, loss = 0.34971806
Iteration 17, loss = 0.36451732
Iteration 18, loss = 0.35039391
Iteration 19, loss = 0.35728753
Iteration 20, loss = 0.34604389
Iteration 21, loss = 0.35439525
Iteration 22, loss = 0.35933804
Iteration 23, loss = 0.36682935
Iteration 24, loss = 0.35321638
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.7025706

Iteration 22, loss = 0.32825532
Iteration 23, loss = 0.32816173
Iteration 24, loss = 0.32728518
Iteration 25, loss = 0.33559734
Iteration 26, loss = 0.32385832
Iteration 27, loss = 0.32395342
Iteration 28, loss = 0.31944456
Iteration 29, loss = 0.32477570
Iteration 30, loss = 0.33339812
Iteration 31, loss = 0.33170869
Iteration 32, loss = 0.32074281
Iteration 33, loss = 0.32037838
Iteration 34, loss = 0.32493809
Iteration 35, loss = 0.32069330
Iteration 36, loss = 0.33130200
Iteration 37, loss = 0.32708716
Iteration 38, loss = 0.32578229
Iteration 39, loss = 0.33257016
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.72320362
Iteration 2, loss = 0.44795640
Iteration 3, loss = 0.39470098
Iteration 4, loss = 0.36165479
Iteration 5, loss = 0.35475526
Iteration 6, loss = 0.34619925
Iteration 7, loss = 0.34923813
Iteration 8, loss = 0.34679401
Iteration 9, loss = 0.33901204
Iteration 10, loss = 0.33598850
Iteration 11, loss = 0.

Iteration 19, loss = 0.30671529
Iteration 20, loss = 0.30612010
Iteration 21, loss = 0.31041708
Iteration 22, loss = 0.30063388
Iteration 23, loss = 0.30652695
Iteration 24, loss = 0.30859445
Iteration 25, loss = 0.31825410
Iteration 26, loss = 0.30947659
Iteration 27, loss = 0.30075974
Iteration 28, loss = 0.31253586
Iteration 29, loss = 0.29910598
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.66715396
Iteration 2, loss = 0.44389492
Iteration 3, loss = 0.38862998
Iteration 4, loss = 0.36338791
Iteration 5, loss = 0.35530034
Iteration 6, loss = 0.34461990
Iteration 7, loss = 0.33970231
Iteration 8, loss = 0.34372999
Iteration 9, loss = 0.33780397
Iteration 10, loss = 0.33203126
Iteration 11, loss = 0.33293485
Iteration 12, loss = 0.32510759
Iteration 13, loss = 0.33060413
Iteration 14, loss = 0.33960335
Iteration 15, loss = 0.34407843
Iteration 16, loss = 0.34258786
Iteration 17, loss = 0.32447356
Iteration 18, loss = 0.

Iteration 39, loss = 0.34893163
Iteration 40, loss = 0.35405159
Iteration 41, loss = 0.34825989
Iteration 42, loss = 0.33256403
Iteration 43, loss = 0.33316935
Iteration 44, loss = 0.32593528
Iteration 45, loss = 0.32343291
Iteration 46, loss = 0.33361443
Iteration 47, loss = 0.33110737
Iteration 48, loss = 0.32510549
Iteration 49, loss = 0.32203347
Iteration 50, loss = 0.33561698
Iteration 51, loss = 0.32506500
Iteration 52, loss = 0.33903427
Iteration 53, loss = 0.33132289
Iteration 54, loss = 0.34052658
Iteration 55, loss = 0.32910273
Iteration 56, loss = 0.32577730
Iteration 57, loss = 0.32410722
Iteration 58, loss = 0.32319044
Iteration 59, loss = 0.32437582
Iteration 60, loss = 0.32785597
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.71006166
Iteration 2, loss = 0.46551607
Iteration 3, loss = 0.41714821
Iteration 4, loss = 0.39185914
Iteration 5, loss = 0.37086664
Iteration 6, loss = 0.35385611
Iteration 7, loss = 

Iteration 17, loss = 0.34318005
Iteration 18, loss = 0.34808751
Iteration 19, loss = 0.33864702
Iteration 20, loss = 0.34395015
Iteration 21, loss = 0.34322393
Iteration 22, loss = 0.35823384
Iteration 23, loss = 0.34410414
Iteration 24, loss = 0.33538826
Iteration 25, loss = 0.33590087
Iteration 26, loss = 0.33375285
Iteration 27, loss = 0.33603062
Iteration 28, loss = 0.34482229
Iteration 29, loss = 0.35396431
Iteration 30, loss = 0.36239205
Iteration 31, loss = 0.37763032
Iteration 32, loss = 0.37930438
Iteration 33, loss = 0.36170423
Iteration 34, loss = 0.34226503
Iteration 35, loss = 0.34758618
Iteration 36, loss = 0.34855625
Iteration 37, loss = 0.34245169
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.75481301
Iteration 2, loss = 0.47432679
Iteration 3, loss = 0.40992591
Iteration 4, loss = 0.38784121
Iteration 5, loss = 0.36456679
Iteration 6, loss = 0.36819643
Iteration 7, loss = 0.35769642
Iteration 8, loss = 0

Iteration 14, loss = 0.33325618
Iteration 15, loss = 0.32921898
Iteration 16, loss = 0.32767708
Iteration 17, loss = 0.32621586
Iteration 18, loss = 0.32510065
Iteration 19, loss = 0.32286193
Iteration 20, loss = 0.34127833
Iteration 21, loss = 0.36598106
Iteration 22, loss = 0.34442862
Iteration 23, loss = 0.33914842
Iteration 24, loss = 0.32605634
Iteration 25, loss = 0.32120899
Iteration 26, loss = 0.34190158
Iteration 27, loss = 0.34639139
Iteration 28, loss = 0.32828626
Iteration 29, loss = 0.32670029
Iteration 30, loss = 0.32460910
Iteration 31, loss = 0.32238842
Iteration 32, loss = 0.33062561
Iteration 33, loss = 0.33789847
Iteration 34, loss = 0.32311775
Iteration 35, loss = 0.34969470
Iteration 36, loss = 0.33986462
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.75998536
Iteration 2, loss = 0.48448928
Iteration 3, loss = 0.42291456
Iteration 4, loss = 0.37725686
Iteration 5, loss = 0.35120845
Iteration 6, loss =

Iteration 27, loss = 0.32259415
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.62571960
Iteration 2, loss = 0.43304940
Iteration 3, loss = 0.37696100
Iteration 4, loss = 0.35557620
Iteration 5, loss = 0.34286573
Iteration 6, loss = 0.34399142
Iteration 7, loss = 0.33375585
Iteration 8, loss = 0.33809422
Iteration 9, loss = 0.34886877
Iteration 10, loss = 0.35260616
Iteration 11, loss = 0.33212310
Iteration 12, loss = 0.32728286
Iteration 13, loss = 0.32602730
Iteration 14, loss = 0.32724704
Iteration 15, loss = 0.32505739
Iteration 16, loss = 0.32054778
Iteration 17, loss = 0.33127421
Iteration 18, loss = 0.33940183
Iteration 19, loss = 0.35149913
Iteration 20, loss = 0.32780914
Iteration 21, loss = 0.32896545
Iteration 22, loss = 0.32101885
Iteration 23, loss = 0.33043235
Iteration 24, loss = 0.32451051
Iteration 25, loss = 0.33610748
Iteration 26, loss = 0.32548222
Iteration 27, loss = 0.33002495
Training loss did not i

Iteration 32, loss = 0.34251465
Iteration 33, loss = 0.34316854
Iteration 34, loss = 0.35300763
Iteration 35, loss = 0.34376184
Iteration 36, loss = 0.34474922
Iteration 37, loss = 0.34751943
Iteration 38, loss = 0.33754311
Iteration 39, loss = 0.33884085
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.65975473
Iteration 2, loss = 0.43044556
Iteration 3, loss = 0.39105180
Iteration 4, loss = 0.36552911
Iteration 5, loss = 0.35577783
Iteration 6, loss = 0.33796783
Iteration 7, loss = 0.34354915
Iteration 8, loss = 0.34948186
Iteration 9, loss = 0.34911639
Iteration 10, loss = 0.34222754
Iteration 11, loss = 0.34811874
Iteration 12, loss = 0.37057378
Iteration 13, loss = 0.35269230
Iteration 14, loss = 0.34217439
Iteration 15, loss = 0.33316232
Iteration 16, loss = 0.33425153
Iteration 17, loss = 0.35446720
Iteration 18, loss = 0.34898422
Iteration 19, loss = 0.34142239
Iteration 20, loss = 0.34466092
Iteration 21, loss = 0.

Iteration 8, loss = 0.36860895
Iteration 9, loss = 0.35868544
Iteration 10, loss = 0.36704109
Iteration 11, loss = 0.37453590
Iteration 12, loss = 0.38306017
Iteration 13, loss = 0.41137867
Iteration 14, loss = 0.37432011
Iteration 15, loss = 0.36044796
Iteration 16, loss = 0.37402438
Iteration 17, loss = 0.35549274
Iteration 18, loss = 0.34461336
Iteration 19, loss = 0.35568938
Iteration 20, loss = 0.34965504
Iteration 21, loss = 0.34715904
Iteration 22, loss = 0.35451789
Iteration 23, loss = 0.34757180
Iteration 24, loss = 0.34153527
Iteration 25, loss = 0.34479320
Iteration 26, loss = 0.35171893
Iteration 27, loss = 0.34291359
Iteration 28, loss = 0.35232229
Iteration 29, loss = 0.36365366
Iteration 30, loss = 0.36644186
Iteration 31, loss = 0.34479776
Iteration 32, loss = 0.34226605
Iteration 33, loss = 0.33705627
Iteration 34, loss = 0.35187003
Iteration 35, loss = 0.36436025
Iteration 36, loss = 0.37143837
Iteration 37, loss = 0.34577618
Iteration 38, loss = 0.35366047
Iteration 

Iteration 20, loss = 0.32846089
Iteration 21, loss = 0.32396246
Iteration 22, loss = 0.32901644
Iteration 23, loss = 0.32606021
Iteration 24, loss = 0.33189275
Iteration 25, loss = 0.32223831
Iteration 26, loss = 0.32219487
Iteration 27, loss = 0.33021097
Iteration 28, loss = 0.33109513
Iteration 29, loss = 0.36327576
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.71791560
Iteration 2, loss = 0.46384311
Iteration 3, loss = 0.39938528
Iteration 4, loss = 0.39335960
Iteration 5, loss = 0.37350474
Iteration 6, loss = 0.35047656
Iteration 7, loss = 0.34138000
Iteration 8, loss = 0.34017372
Iteration 9, loss = 0.34458355
Iteration 10, loss = 0.34408107
Iteration 11, loss = 0.33094582
Iteration 12, loss = 0.33955658
Iteration 13, loss = 0.33676690
Iteration 14, loss = 0.34677842
Iteration 15, loss = 0.34559650
Iteration 16, loss = 0.32947137
Iteration 17, loss = 0.33077199
Iteration 18, loss = 0.33218052
Iteration 19, loss = 0.

Iteration 11, loss = 0.33023812
Iteration 12, loss = 0.33510388
Iteration 13, loss = 0.33427880
Iteration 14, loss = 0.32469364
Iteration 15, loss = 0.32928290
Iteration 16, loss = 0.32679208
Iteration 17, loss = 0.32929311
Iteration 18, loss = 0.32847899
Iteration 19, loss = 0.33457527
Iteration 20, loss = 0.34032858
Iteration 21, loss = 0.33731531
Iteration 22, loss = 0.33442505
Iteration 23, loss = 0.33481938
Iteration 24, loss = 0.34089904
Iteration 25, loss = 0.33549833
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.76653061
Iteration 2, loss = 0.48508791
Iteration 3, loss = 0.40537375
Iteration 4, loss = 0.38599435
Iteration 5, loss = 0.35025961
Iteration 6, loss = 0.33979106
Iteration 7, loss = 0.33531199
Iteration 8, loss = 0.34059454
Iteration 9, loss = 0.33452287
Iteration 10, loss = 0.33549414
Iteration 11, loss = 0.33175026
Iteration 12, loss = 0.33211483
Iteration 13, loss = 0.33543885
Iteration 14, loss = 0.

Iteration 15, loss = 0.33302490
Iteration 16, loss = 0.32525491
Iteration 17, loss = 0.32759694
Iteration 18, loss = 0.33447117
Iteration 19, loss = 0.33281260
Iteration 20, loss = 0.33193653
Iteration 21, loss = 0.33769513
Iteration 22, loss = 0.32471643
Iteration 23, loss = 0.32554912
Iteration 24, loss = 0.32083415
Iteration 25, loss = 0.32015161
Iteration 26, loss = 0.31761642
Iteration 27, loss = 0.33029736
Iteration 28, loss = 0.32514132
Iteration 29, loss = 0.32923790
Iteration 30, loss = 0.33628606
Iteration 31, loss = 0.32899962
Iteration 32, loss = 0.32680080
Iteration 33, loss = 0.32885681
Iteration 34, loss = 0.32762493
Iteration 35, loss = 0.32998370
Iteration 36, loss = 0.31812119
Iteration 37, loss = 0.32058118
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.69170201
Iteration 2, loss = 0.42599201
Iteration 3, loss = 0.38254249
Iteration 4, loss = 0.35944545
Iteration 5, loss = 0.34316603
Iteration 6, loss =

Iteration 10, loss = 0.34563028
Iteration 11, loss = 0.35038816
Iteration 12, loss = 0.36913149
Iteration 13, loss = 0.35186454
Iteration 14, loss = 0.33908068
Iteration 15, loss = 0.35046030
Iteration 16, loss = 0.33916609
Iteration 17, loss = 0.33760096
Iteration 18, loss = 0.34271846
Iteration 19, loss = 0.35084819
Iteration 20, loss = 0.33747714
Iteration 21, loss = 0.33090964
Iteration 22, loss = 0.33555061
Iteration 23, loss = 0.33429029
Iteration 24, loss = 0.33647775
Iteration 25, loss = 0.34763064
Iteration 26, loss = 0.33983515
Iteration 27, loss = 0.33005820
Iteration 28, loss = 0.33918381
Iteration 29, loss = 0.35050079
Iteration 30, loss = 0.33096251
Iteration 31, loss = 0.33035084
Iteration 32, loss = 0.33714969
Iteration 33, loss = 0.33820845
Iteration 34, loss = 0.33336973
Iteration 35, loss = 0.32706281
Iteration 36, loss = 0.33097675
Iteration 37, loss = 0.32509690
Iteration 38, loss = 0.33716927
Iteration 39, loss = 0.32333752
Iteration 40, loss = 0.32613162
Iteratio

Iteration 36, loss = 0.33556520
Iteration 37, loss = 0.33534908
Iteration 38, loss = 0.34654715
Iteration 39, loss = 0.33382022
Iteration 40, loss = 0.32774064
Iteration 41, loss = 0.32925885
Iteration 42, loss = 0.32209856
Iteration 43, loss = 0.33325938
Iteration 44, loss = 0.32835310
Iteration 45, loss = 0.32872031
Iteration 46, loss = 0.33200898
Iteration 47, loss = 0.33302699
Iteration 48, loss = 0.32072251
Iteration 49, loss = 0.33702746
Iteration 50, loss = 0.33152036
Iteration 51, loss = 0.33584010
Iteration 52, loss = 0.34117921
Iteration 53, loss = 0.32625631
Iteration 54, loss = 0.34059504
Iteration 55, loss = 0.33481722
Iteration 56, loss = 0.34723525
Iteration 57, loss = 0.33431110
Iteration 58, loss = 0.34095872
Iteration 59, loss = 0.33097386
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.82890712
Iteration 2, loss = 0.47734043
Iteration 3, loss = 0.42034469
Iteration 4, loss = 0.38883980
Iteration 5, loss 

Iteration 14, loss = 0.36222953
Iteration 15, loss = 0.36081312
Iteration 16, loss = 0.35600732
Iteration 17, loss = 0.34890648
Iteration 18, loss = 0.35660069
Iteration 19, loss = 0.34663761
Iteration 20, loss = 0.34018615
Iteration 21, loss = 0.33790271
Iteration 22, loss = 0.33515267
Iteration 23, loss = 0.33920165
Iteration 24, loss = 0.35552221
Iteration 25, loss = 0.34649025
Iteration 26, loss = 0.33423224
Iteration 27, loss = 0.34229920
Iteration 28, loss = 0.35903727
Iteration 29, loss = 0.37307263
Iteration 30, loss = 0.36438885
Iteration 31, loss = 0.36203562
Iteration 32, loss = 0.34299163
Iteration 33, loss = 0.34041212
Iteration 34, loss = 0.33867915
Iteration 35, loss = 0.35141683
Iteration 36, loss = 0.34408880
Iteration 37, loss = 0.35876383
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.75340976
Iteration 2, loss = 0.49053822
Iteration 3, loss = 0.46248500
Iteration 4, loss = 0.40345818
Iteration 5, loss 

Iteration 42, loss = 0.31931019
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.70562205
Iteration 2, loss = 0.45337393
Iteration 3, loss = 0.39415723
Iteration 4, loss = 0.36221331
Iteration 5, loss = 0.35392682
Iteration 6, loss = 0.33941649
Iteration 7, loss = 0.34034702
Iteration 8, loss = 0.33553620
Iteration 9, loss = 0.34527800
Iteration 10, loss = 0.32494010
Iteration 11, loss = 0.32110216
Iteration 12, loss = 0.32456353
Iteration 13, loss = 0.33381453
Iteration 14, loss = 0.33769169
Iteration 15, loss = 0.33315576
Iteration 16, loss = 0.33966620
Iteration 17, loss = 0.33021858
Iteration 18, loss = 0.35289137
Iteration 19, loss = 0.36159449
Iteration 20, loss = 0.34479333
Iteration 21, loss = 0.36174541
Iteration 22, loss = 0.34031358
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.72193073
Iteration 2, loss = 0.48637779
Iteration 3, loss = 0.40491876


Iteration 31, loss = 0.29893413
Iteration 32, loss = 0.30145431
Iteration 33, loss = 0.29892489
Iteration 34, loss = 0.31587312
Iteration 35, loss = 0.30941660
Iteration 36, loss = 0.31316221
Iteration 37, loss = 0.30065035
Iteration 38, loss = 0.30357943
Iteration 39, loss = 0.30660315
Iteration 40, loss = 0.31937180
Iteration 41, loss = 0.32254445
Iteration 42, loss = 0.32452088
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.73358030
Iteration 2, loss = 0.48419056
Iteration 3, loss = 0.40010661
Iteration 4, loss = 0.36199218
Iteration 5, loss = 0.34482894
Iteration 6, loss = 0.33868941
Iteration 7, loss = 0.33803427
Iteration 8, loss = 0.32921329
Iteration 9, loss = 0.33523418
Iteration 10, loss = 0.33725628
Iteration 11, loss = 0.33253424
Iteration 12, loss = 0.33433300
Iteration 13, loss = 0.34695557
Iteration 14, loss = 0.33165549
Iteration 15, loss = 0.33300981
Iteration 16, loss = 0.32805419
Iteration 17, loss = 0.

Iteration 23, loss = 0.35045610
Iteration 24, loss = 0.34951305
Iteration 25, loss = 0.33969691
Iteration 26, loss = 0.34136549
Iteration 27, loss = 0.34333509
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.79111467
Iteration 2, loss = 0.48795231
Iteration 3, loss = 0.42419442
Iteration 4, loss = 0.38071326
Iteration 5, loss = 0.35958010
Iteration 6, loss = 0.35165892
Iteration 7, loss = 0.34501451
Iteration 8, loss = 0.34425871
Iteration 9, loss = 0.36172397
Iteration 10, loss = 0.35274081
Iteration 11, loss = 0.36696298
Iteration 12, loss = 0.35972425
Iteration 13, loss = 0.37136728
Iteration 14, loss = 0.35268908
Iteration 15, loss = 0.34443840
Iteration 16, loss = 0.33219288
Iteration 17, loss = 0.34080931
Iteration 18, loss = 0.33793332
Iteration 19, loss = 0.34522505
Iteration 20, loss = 0.33360648
Iteration 21, loss = 0.33471782
Iteration 22, loss = 0.34122477
Iteration 23, loss = 0.33148372
Iteration 24, loss = 0.

Iteration 19, loss = 0.35243767
Iteration 20, loss = 0.35035242
Iteration 21, loss = 0.34934892
Iteration 22, loss = 0.35655991
Iteration 23, loss = 0.35853329
Iteration 24, loss = 0.34756017
Iteration 25, loss = 0.38280123
Iteration 26, loss = 0.35852394
Iteration 27, loss = 0.35557793
Iteration 28, loss = 0.34273599
Iteration 29, loss = 0.35632368
Iteration 30, loss = 0.37576319
Iteration 31, loss = 0.36011528
Iteration 32, loss = 0.35387818
Iteration 33, loss = 0.35279357
Iteration 34, loss = 0.35075394
Iteration 35, loss = 0.35137570
Iteration 36, loss = 0.36211481
Iteration 37, loss = 0.36944617
Iteration 38, loss = 0.34838324
Iteration 39, loss = 0.35097187
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.75545410
Iteration 2, loss = 0.48179167
Iteration 3, loss = 0.41936297
Iteration 4, loss = 0.37681128
Iteration 5, loss = 0.35116742
Iteration 6, loss = 0.34400657
Iteration 7, loss = 0.35307959
Iteration 8, loss = 0

Iteration 13, loss = 0.34907546
Iteration 14, loss = 0.32856274
Iteration 15, loss = 0.34025497
Iteration 16, loss = 0.35357123
Iteration 17, loss = 0.34908399
Iteration 18, loss = 0.34217154
Iteration 19, loss = 0.33085503
Iteration 20, loss = 0.34231450
Iteration 21, loss = 0.33867708
Iteration 22, loss = 0.34563803
Iteration 23, loss = 0.33911252
Iteration 24, loss = 0.34519598
Iteration 25, loss = 0.35302795
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.84334419
Iteration 2, loss = 0.50215012
Iteration 3, loss = 0.42992159
Iteration 4, loss = 0.39892819
Iteration 5, loss = 0.37803852
Iteration 6, loss = 0.35640284
Iteration 7, loss = 0.35454160
Iteration 8, loss = 0.35513797
Iteration 9, loss = 0.34880400
Iteration 10, loss = 0.34235406
Iteration 11, loss = 0.34776547
Iteration 12, loss = 0.35651426
Iteration 13, loss = 0.34450888
Iteration 14, loss = 0.33978310
Iteration 15, loss = 0.35845052
Iteration 16, loss = 0.

Iteration 34, loss = 0.33473208
Iteration 35, loss = 0.32848615
Iteration 36, loss = 0.33335504
Iteration 37, loss = 0.33023179
Iteration 38, loss = 0.32439598
Iteration 39, loss = 0.32799377
Iteration 40, loss = 0.32486107
Iteration 41, loss = 0.33539901
Iteration 42, loss = 0.32605669
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.73720138
Iteration 2, loss = 0.49792718
Iteration 3, loss = 0.41561333
Iteration 4, loss = 0.40011998
Iteration 5, loss = 0.35942466
Iteration 6, loss = 0.34629189
Iteration 7, loss = 0.33989190
Iteration 8, loss = 0.33078721
Iteration 9, loss = 0.32909582
Iteration 10, loss = 0.32480639
Iteration 11, loss = 0.32866828
Iteration 12, loss = 0.33438665
Iteration 13, loss = 0.33017100
Iteration 14, loss = 0.33256096
Iteration 15, loss = 0.34679200
Iteration 16, loss = 0.32897237
Iteration 17, loss = 0.32867903
Iteration 18, loss = 0.32773918
Iteration 19, loss = 0.32260630
Iteration 20, loss = 0.

Iteration 21, loss = 0.35015829
Iteration 22, loss = 0.34575155
Iteration 23, loss = 0.34410652
Iteration 24, loss = 0.35759121
Iteration 25, loss = 0.34995977
Iteration 26, loss = 0.34411818
Iteration 27, loss = 0.34077456
Iteration 28, loss = 0.34321522
Iteration 29, loss = 0.34245125
Iteration 30, loss = 0.33892465
Iteration 31, loss = 0.34878107
Iteration 32, loss = 0.35827043
Iteration 33, loss = 0.35669388
Iteration 34, loss = 0.35727338
Iteration 35, loss = 0.36594492
Iteration 36, loss = 0.35279561
Iteration 37, loss = 0.33764467
Iteration 38, loss = 0.33519477
Iteration 39, loss = 0.33766440
Iteration 40, loss = 0.34820524
Iteration 41, loss = 0.34479270
Iteration 42, loss = 0.34445738
Iteration 43, loss = 0.35205025
Iteration 44, loss = 0.36249249
Iteration 45, loss = 0.34484596
Iteration 46, loss = 0.33972629
Iteration 47, loss = 0.36376375
Iteration 48, loss = 0.35306811
Iteration 49, loss = 0.36404020
Training loss did not improve more than tol=0.000100 for 10 consecutive 

Iteration 22, loss = 0.33842912
Iteration 23, loss = 0.34923369
Iteration 24, loss = 0.34185098
Iteration 25, loss = 0.34270661
Iteration 26, loss = 0.33844464
Iteration 27, loss = 0.34255049
Iteration 28, loss = 0.34860339
Iteration 29, loss = 0.35913515
Iteration 30, loss = 0.35974604
Iteration 31, loss = 0.34900924
Iteration 32, loss = 0.34319267
Iteration 33, loss = 0.33377119
Iteration 34, loss = 0.33687989
Iteration 35, loss = 0.34824755
Iteration 36, loss = 0.34829221
Iteration 37, loss = 0.34569966
Iteration 38, loss = 0.34066081
Iteration 39, loss = 0.34376673
Iteration 40, loss = 0.34497215
Iteration 41, loss = 0.33870441
Iteration 42, loss = 0.34715567
Iteration 43, loss = 0.35310800
Iteration 44, loss = 0.33931814
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.82045612
Iteration 2, loss = 0.49397507
Iteration 3, loss = 0.41844295
Iteration 4, loss = 0.39441331
Iteration 5, loss = 0.36822432
Iteration 6, loss =

Iteration 11, loss = 0.33427103
Iteration 12, loss = 0.33677317
Iteration 13, loss = 0.34993718
Iteration 14, loss = 0.33753377
Iteration 15, loss = 0.33844476
Iteration 16, loss = 0.33773108
Iteration 17, loss = 0.33493982
Iteration 18, loss = 0.33778582
Iteration 19, loss = 0.34194589
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.85323889
Iteration 2, loss = 0.51679578
Iteration 3, loss = 0.43588986
Iteration 4, loss = 0.38641472
Iteration 5, loss = 0.36690302
Iteration 6, loss = 0.35663994
Iteration 7, loss = 0.35510308
Iteration 8, loss = 0.34071574
Iteration 9, loss = 0.34114744
Iteration 10, loss = 0.34169009
Iteration 11, loss = 0.34255972
Iteration 12, loss = 0.35170661
Iteration 13, loss = 0.34869294
Iteration 14, loss = 0.34107776
Iteration 15, loss = 0.33695741
Iteration 16, loss = 0.34928432
Iteration 17, loss = 0.35959985
Iteration 18, loss = 0.35223262
Iteration 19, loss = 0.37106783
Iteration 20, loss = 0.

Iteration 23, loss = 0.30607820
Iteration 24, loss = 0.30636603
Iteration 25, loss = 0.29815465
Iteration 26, loss = 0.30485012
Iteration 27, loss = 0.30175211
Iteration 28, loss = 0.29743804
Iteration 29, loss = 0.31517258
Iteration 30, loss = 0.32896097
Iteration 31, loss = 0.30911596
Iteration 32, loss = 0.31805435
Iteration 33, loss = 0.33617078
Iteration 34, loss = 0.32229749
Iteration 35, loss = 0.31249033
Iteration 36, loss = 0.29994367
Iteration 37, loss = 0.30098170
Iteration 38, loss = 0.30285820
Iteration 39, loss = 0.30577047
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.80060266
Iteration 2, loss = 0.49072241
Iteration 3, loss = 0.41291093
Iteration 4, loss = 0.37072818
Iteration 5, loss = 0.34865700
Iteration 6, loss = 0.33842743
Iteration 7, loss = 0.33779939
Iteration 8, loss = 0.34411827
Iteration 9, loss = 0.33253702
Iteration 10, loss = 0.33075643
Iteration 11, loss = 0.32980206
Iteration 12, loss = 0.

Iteration 27, loss = 0.36449617
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.79365333
Iteration 2, loss = 0.50050157
Iteration 3, loss = 0.42220795
Iteration 4, loss = 0.38043596
Iteration 5, loss = 0.37240799
Iteration 6, loss = 0.36818441
Iteration 7, loss = 0.35405796
Iteration 8, loss = 0.33976932
Iteration 9, loss = 0.34035614
Iteration 10, loss = 0.33236979
Iteration 11, loss = 0.34027559
Iteration 12, loss = 0.34094799
Iteration 13, loss = 0.35047219
Iteration 14, loss = 0.35762546
Iteration 15, loss = 0.33596383
Iteration 16, loss = 0.33587695
Iteration 17, loss = 0.34175113
Iteration 18, loss = 0.35950542
Iteration 19, loss = 0.35530327
Iteration 20, loss = 0.34792054
Iteration 21, loss = 0.36378985
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.80647769
Iteration 2, loss = 0.48420541
Iteration 3, loss = 0.40894983
Iteration 4, loss = 0.34792838
I

Iteration 23, loss = 0.35027471
Iteration 24, loss = 0.35293254
Iteration 25, loss = 0.35032993
Iteration 26, loss = 0.36059465
Iteration 27, loss = 0.36099912
Iteration 28, loss = 0.35048013
Iteration 29, loss = 0.33802201
Iteration 30, loss = 0.36449212
Iteration 31, loss = 0.35065246
Iteration 32, loss = 0.37435647
Iteration 33, loss = 0.35526369
Iteration 34, loss = 0.34189133
Iteration 35, loss = 0.34387038
Iteration 36, loss = 0.34726215
Iteration 37, loss = 0.35019970
Iteration 38, loss = 0.35941370
Iteration 39, loss = 0.38381025
Iteration 40, loss = 0.35930489
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.80074904
Iteration 2, loss = 0.52749668
Iteration 3, loss = 0.42969047
Iteration 4, loss = 0.38086285
Iteration 5, loss = 0.36096257
Iteration 6, loss = 0.34851801
Iteration 7, loss = 0.35282757
Iteration 8, loss = 0.36491986
Iteration 9, loss = 0.37363661
Iteration 10, loss = 0.34398109
Iteration 11, loss = 0.

Iteration 29, loss = 0.33946884
Iteration 30, loss = 0.34110359
Iteration 31, loss = 0.37570884
Iteration 32, loss = 0.36744813
Iteration 33, loss = 0.38274095
Iteration 34, loss = 0.37531567
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.86954971
Iteration 2, loss = 0.49934598
Iteration 3, loss = 0.41963076
Iteration 4, loss = 0.38352602
Iteration 5, loss = 0.38557710
Iteration 6, loss = 0.37522382
Iteration 7, loss = 0.36354453
Iteration 8, loss = 0.36233485
Iteration 9, loss = 0.36632253
Iteration 10, loss = 0.37188431
Iteration 11, loss = 0.36371722
Iteration 12, loss = 0.36689963
Iteration 13, loss = 0.37875496
Iteration 14, loss = 0.36823477
Iteration 15, loss = 0.36770504
Iteration 16, loss = 0.35411000
Iteration 17, loss = 0.36844274
Iteration 18, loss = 0.38993876
Iteration 19, loss = 0.36124501
Iteration 20, loss = 0.35865323
Iteration 21, loss = 0.37059412
Iteration 22, loss = 0.35833042
Iteration 23, loss = 0.

Iteration 25, loss = 0.37401204
Iteration 26, loss = 0.36798245
Iteration 27, loss = 0.34400141
Iteration 28, loss = 0.32674855
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.86431176
Iteration 2, loss = 0.51187587
Iteration 3, loss = 0.41953981
Iteration 4, loss = 0.38062666
Iteration 5, loss = 0.35943908
Iteration 6, loss = 0.35334168
Iteration 7, loss = 0.34656860
Iteration 8, loss = 0.34798747
Iteration 9, loss = 0.33613978
Iteration 10, loss = 0.34311109
Iteration 11, loss = 0.36059526
Iteration 12, loss = 0.36187813
Iteration 13, loss = 0.37455652
Iteration 14, loss = 0.36268419
Iteration 15, loss = 0.35637850
Iteration 16, loss = 0.34483485
Iteration 17, loss = 0.34042856
Iteration 18, loss = 0.34612859
Iteration 19, loss = 0.34350228
Iteration 20, loss = 0.34887543
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.
Iteration 1, loss = 0.83639369
Iteration 2, loss = 0.51982743

[Parallel(n_jobs=1)]: Done 500 out of 500 | elapsed:  2.8min finished


Iteration 10, loss = 0.36029339
Iteration 11, loss = 0.35573617
Iteration 12, loss = 0.34097755
Iteration 13, loss = 0.35471312
Iteration 14, loss = 0.35747684
Iteration 15, loss = 0.34473521
Iteration 16, loss = 0.35064357
Iteration 17, loss = 0.35357976
Iteration 18, loss = 0.34027195
Iteration 19, loss = 0.34908899
Training loss did not improve more than tol=0.000100 for 10 consecutive epochs. Stopping.


GridSearchCV(cv=10,
             estimator=MLPClassifier(activation='logistic', alpha=1,
                                     hidden_layer_sizes=(30,),
                                     learning_rate_init=0.02, random_state=1,
                                     verbose=True),
             param_grid={'hidden_layer_sizes': [(5,), (10,), (15,), (20,),
                                                (25,), (30,), (35,), (40,),
                                                (45,), (50,), (55,), (60,),
                                                (65,), (70,), (75,), (80,),
                                                (85,), (90,), (95,), (100,),
                                                (105,), (110,), (115,), (120,),
                                                (125,), (130,), (135,), (140,),
                                                (145,), (150,), ...]},
             scoring='accuracy', verbose=True)

In [12]:
# finding the best number of neurons
print(grid.best_score_)
print('Best Number of Neurons:')
print(grid.best_params_)

0.8758105235184693
Best Number of Neurons:
{'hidden_layer_sizes': (230,)}


## Answer Summary

## c
[0.84507042 0.78873239 0.88732394 0.90610329 0.87323944 0.90140845
 0.86792453 0.90566038 0.85849057 0.67924528]

## d
Best Number of Neurons:
{'hidden_layer_sizes': (230,)}